### PR TITLE
React/DP-13800:  Deprecated lifecycle methods

### PR DESCRIPTION
**Pull request by smurrayatwork**
## Description
- (React) [InputTextFuzzy] DP-13800: Adds prop onKeyDownPreventDefault. When set to true, hitting enter without a valid suggestion will trigger event.preventDefault().

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-13800)

## Steps to Test
1. Pull the branch and ensure all automated tests pass and InputTextFuzzy has no visual changes in functionality.

### CODE REVIEW

**Review by avrilpearl — APPROVED** [review]

File: react/src/components/atoms/forms/InputTextFuzzy/index.js [hunk 1]
```diff
@@ -261,7 +267,9 @@ InputTextFuzzy.propTypes = {
   /** The id of the the input tag */
   inputId: PropTypes.string,
   /** By default all options will be rendered as suggestions on input focus */
-  renderDefaultSuggestion: PropTypes.bool
+  renderDefaultSuggestion: PropTypes.bool,
+  /** When true, calls preventDefault on the onKeyDown SyntheticEvent for the input element. */
+  onKeyDownPreventDefault: PropTypes.bool
```
Comment (clairesunstudio) [inline react/src/components/atoms/forms/InputTextFuzzy/index.js]:
```suggestion
  onEmptyEnterPreventDefault: PropTypes.bool
```
can we change the name of this prop to be more descriptive of what it's doing?

**Comment by clairesunstudio** [comment]
Can you in this pr, switching InputTextTypeAhead with InputTextFuzzy for pressfilter in filterbox/SearchBanner? so we don't need to npm link to search to test the form submission

**Comment by clairesunstudio** [comment]
filter-clear

In search, currently this is preventing empty value from inputTextFuzzy to be submitted.
The expected behavior is that when the inputTextFuzzy value is empty and user hit enter, the filterBox form will submit it as an empty value. If a filter tag of that field already exist, it will remove that filter tag.

**Review by clairesunstudio — APPROVED** [review]
tested with https://github.com/massgov/massgov-search/pull/353 and works as expected.

**Pull request merged**